src/mesh/generic/Boundary.cpp

```diff
@@ -83,4 +99,73 @@ double Boundary::interpolate_prescribed_load(std::vector<std::pair<double, doubl
 
     return (1.0 - interpolation_factor) * load_0 + interpolation_factor * load_1;
 }
+
+void Boundary::generate_sinusoidal(Json::Value const& boundary, std::string const& name)
+{
+    using namespace ranges;
+
+    Json::Value const& Amplitude = boundary["Amplitude"][name];
+    Json::Value const& Period = boundary["Period"];
+    Json::Value const& NumberOfCycles = boundary["NumberOfCycles"];
+    Json::Value const& TimeStep = boundary["TimeStep"];
+    Json::Value const& Phase = boundary["Phase"];
+
+    std::vector<double> amplitude = Amplitude | view::transform([](auto const& element) {
```

Comment:
Grab the key and use that directly in the range and you can drop the references :)

```cpp
 = boundary["Amplitude"][name] | view::transform(...)
````